### ISSUE TITLE
日本語BARTのfine-tuning

### ISSUE THREAD

**Issue by tomishima2904** [tomishima2904/explore_conceptnet#10]
# 目標
head, tail, headとtailを含む文 → relation
```
# 擬似コード (puseudo code)
head = "ショートカット"
tail = "ヘアスタイル"
sentence = "セミロングのふわっとした女の子らしいヘアスタイルは「聖子ちゃんカット」と呼ばれて全国の若い女性の間で大流行し、翌年1981年の年末に突然、聖子がショートカットになると今度はショートカットが流行り出した"

relation = model(tail, head, sentence)
print(relation)  # の一種である
```

# 実験条件
[京都大学](https://nlp.ist.i.kyoto-u.ac.jp/?BART%E6%97%A5%E6%9C%AC%E8%AA%9EPretrained%E3%83%A2%E3%83%87%E3%83%AB)が公開している訓練済み日本語BART ([hugging face](https://huggingface.co/ku-nlp/bart-base-japanese))

# TODO
- データセットをあらかじめJumanでトークナイズ
- fine-tuningのやり方を知る

**Comment by tomishima2904** [comment]
[これ](https://colab.research.google.com/drive/182fculAfvkSqwYb-QM5eNJmJCBcmiRVJ?usp=sharing#scrollTo=8iqM4VL6dBQh)によると、適切な形式のデータセットを作成しちゃえば、あとは以下のようにするだけぽい？
```
trainer = Seq2SeqTrainer(
    model,
    args,
    train_dataset=AdditionDataset(2,"train"),
    eval_dataset=AdditionDataset(2,"test"),
    data_collator=data_collator,
    tokenizer=tokenizer,
)

trainer.train()
```

`torch.utils.data`の`Dataset`はhugging face用に決まった形式にしないといけないのかも？

**Comment by tomishima2904** [comment]
hugging faceに[tutorial](https://huggingface.co/learn/nlp-course/chapter1/1)があったので勉強中

**Comment by tomishima2904** [comment]
#8 京都大学の日本語訓練済みBARTは入力文をあらかじめ Juman++ でトークナイズしておく必要があるので、それも行う。

```
from transformers import AutoTokenizer, MBartForConditionalGeneration
tokenizer = AutoTokenizer.from_pretrained('ku-nlp/bart-base-japanese')
model = MBartForConditionalGeneration.from_pretrained('ku-nlp/bart-base-japanese')
sentence = '京都 大学 で 自然 言語 処理 を 専攻 する 。'  # input should be segmented into words by Juman++ in advance
encoding = tokenizer(sentence, return_tensors='pt')
...
```

**Comment by tomishima2904** [comment]
## BARTではデータセットをどんな形状・カラム名にすればいいのか
BARTでは`BartForConditionalGeneration`が元になっているのでこのモデルタイプが受け入れるデータセットの形状を知りたい

[ここ](https://discuss.huggingface.co/t/train-bart-for-conditional-generation-e-g-summarization/1904)のフォーラムによると、`columns = ['input_ids', 'labels', 'decoder_input_ids','attention_mask',] `とすれば良さそう。
- `input_ids`: （おそらくpaddingされた）入力文
- `labels`: 自分の実装ではいらないと思われる
- `decoder_input_ids`: ターゲット文（自分の場合はrelation)
- `attention_mask`: `input_ids`がpaddingされているところを判断

**Comment by tomishima2904** [comment]
## ↑のような回りくどいことはしなくていい！
京都大学のbartのGithubリポジトリの[readme](https://github.com/utanaka2000/fairseq/blob/japanese_bart_pretrained_model/JAPANESE_BART_README.md)にやるべきことがちゃんと書いてあった
